sympy/functions/combinatorial/factorials.py

```diff
@@ -868,192 +922,304 @@ def fdiff(self, argindex=1):
             raise ArgumentIndexError(self, argindex)
 
     @classmethod
-    def _eval(self, n, k):
+    def _eval(cls, n, k):
         # n.is_Number and k.is_Integer and k != 1 and n != k
-
-        if k.is_Integer:
-            if n.is_Integer and n >= 0:
-                n, k = int(n), int(k)
-
-                if k > n:
-                    return S.Zero
-                elif k > n // 2:
-                    k = n - k
-
-                if HAS_GMPY:
-                    from sympy.core.compatibility import gmpy
-                    return Integer(gmpy.bincoef(n, k))
-
-                d, result = n - k, 1
-                for i in range(1, k + 1):
-                    d += 1
-                    result = result * d // i
-                return Integer(result)
-            else:
-                d, result = n - k, 1
-                for i in range(1, k + 1):
-                    d += 1
-                    result *= d
-                    result /= i
-                return result
+        # and k > 0 (and, if n is positive, k < n - k)
+        Ntype = Float if k.has(Float) or n.has(Float) else Integer
+        k = as_int(k, strict=False)
+        direct = n.is_nonnegative
+        if direct:
+            try:
+                n = as_int(n, strict=False)
+            except ValueError:
+                direct = False
+        if direct:
+
+            if HAS_GMPY:
+                from sympy.core.compatibility import gmpy
+                return Ntype(gmpy.bincoef(n, k))
+
+            d, result = n - k, 1
+            for i in range(1, k + 1):
+                d += 1
+                result = result * d // i
+            return Ntype(result)
+        else:
+            return Mul(*[n + i for i in range(1 - k, 1)])/factorial(k)
```

Comment:
![Screen Shot 2020-04-22 at 1 56 45 AM](https://user-images.githubusercontent.com/28682735/79910789-ebf7a100-843c-11ea-92d9-5390b844b585.png)
[link here](https://en.wikipedia.org/wiki/Binomial_coefficient#Binomial_coefficients_as_polynomials)
This is what our implementation does.
```python
>>> n = -3; k = -5
>>> ff(n,k)/factorial(k)
nan
>>> k = n - k
>>> ff(n,k)/factorial(k) <- we choose the second option
6
```
and yes [`binomial(-3, -5) == 6`](http://www.wolframalpha.com/input/?i=Binomial%5B-3%2C%20-5%5D)